web/src/user_topics.ts

```diff
@@ -38,41 +69,41 @@ export function update_user_topics(stream_id, topic, visibility_policy, date_upd
     }
 }
 
-export function get_topic_visibility_policy(stream_id, topic) {
+export function get_topic_visibility_policy(stream_id: number, topic: string): number | boolean {
     if (stream_id === undefined) {
         return false;
     }
     const sub_dict = all_user_topics.get(stream_id);
     if (sub_dict && sub_dict.get(topic)) {
-        return sub_dict.get(topic).visibility_policy;
+        return sub_dict.get(topic)!.visibility_policy;
     }
 
     return all_visibility_policies.INHERIT;
 }
 
-export function is_topic_followed(stream_id, topic) {
+export function is_topic_followed(stream_id: number, topic: string): boolean {
     return get_topic_visibility_policy(stream_id, topic) === all_visibility_policies.FOLLOWED;
 }
 
-export function is_topic_unmuted(stream_id, topic) {
+export function is_topic_unmuted(stream_id: number, topic: string): boolean {
     return get_topic_visibility_policy(stream_id, topic) === all_visibility_policies.UNMUTED;
 }
 
-export function is_topic_muted(stream_id, topic) {
+export function is_topic_muted(stream_id: number, topic: string): boolean {
     return get_topic_visibility_policy(stream_id, topic) === all_visibility_policies.MUTED;
 }
 
-export function is_topic_unmuted_or_followed(stream_id, topic) {
+export function is_topic_unmuted_or_followed(stream_id: number, topic: string): boolean {
     return is_topic_unmuted(stream_id, topic) || is_topic_followed(stream_id, topic);
 }
 
-export function get_user_topics_for_visibility_policy(visibility_policy) {
-    const topics = [];
+export function get_user_topics_for_visibility_policy(visibility_policy: number): UserTopic[] {
+    const topics: UserTopic[] = [];
     for (const [stream_id, sub_dict] of all_user_topics) {
         const stream = sub_store.maybe_get_stream_name(stream_id);
         for (const topic of sub_dict.keys()) {
-            if (sub_dict.get(topic).visibility_policy === visibility_policy) {
-                const date_updated = sub_dict.get(topic).date_updated;
+            if (sub_dict.get(topic)!.visibility_policy === visibility_policy) {
+                const date_updated = sub_dict.get(topic)!.date_updated;
                 const date_updated_str = timerender.render_now(new Date(date_updated)).time_str;
```

Comment:
Similarly this calculation might make sense to move to the `modified` function; @prakhar1144 FYI.